content/docs/dvclive/ml-frameworks/pytorch-lightning.md

```diff
@@ -0,0 +1,69 @@
+# PyTorch Lightning
+
+DVCLive allows you to easily add experiment tracking capabilities to your
+PyTorch Lightning projects.
+
+## About PyTorch Lightning
+
+[PyTorch Lightning](https://www.pytorchlightning.ai/) is an open-source
+framework for training PyTorch networks.
+
+## Usage
+
+To start using DVCLive you just need to add a few lines to your training code in
+**any** PyTorch Lightning project.
+
+You just need to pass the
+[`DvcLiveLogger`](https://github.com/iterative/dvclive/blob/master/dvclive/lightning.py)
+to your `trainer`:
+
+```git
++from dvclive.lightning import DvcLiveLogger
+
+. . .
+ dvclive_logger = DvcLiveLogger()
+
+ trainer = Trainer(
++       logger=dvclive_logger,
+    )
+ trainer.fit(model)
+```
+
+This will generate the metrics logs and summaries as described in the
+[Get Started](/docs/dvclive/get-started#outputs).
+
+> 💡Without requiring additional modifications to your training code, you can
+> use DVCLive alongside DVC. See
+> [DVCLive with DVC](/doc/dvclive/dvclive-with-dvc) for more info.
+
+## Parameters
+
+- `run_name` - Optional name of the run, used in PyTorch Lightning to get
+  version.
+- `prefix` - Optional string that adds to each metric name.
+- `experiment` - Optional DVCLive object which will be used instead of
+  initializing a new one.
+
+You can also pass any argument of dvclive object described in the
+[dvclive.init()](/docs/dvclive/api-reference/init).
+
+Example:
+
+```python
+from dvclive.lightning import DvcLiveLogger
+
+dvclive_logger = DvcLiveLogger(
+    path='my_logs_path',
+    summary=False
+)
+trainer = Trainer(
+    logger=dvclive_logger,
+)
+trainer.fit(model)
+```
+
+> 📖 PyTorch Lightning by default creates directory with the logger's name to
```

Comment:
This reminded me that we should include logic for saving the model inside `DvcLiveLogger` to be consistent with the other integrations (which support the `model_file` arg).

Reply:
https://github.com/iterative/dvclive/issues/170

Reply:
In Lightning saving is controlled by another callback and is independent of logger. I suppose, we can't include this logic and keep compatibility with native pytorch lightning checkpoint callbacks

Reply:
`LightningModule` implements method `log`, which user call to store everything he wants, and then `LightningModule` calls logger. Checkpoints are saved at the end of batch/epoch via hooks like `on_train_batch_end`, etc. Logger doesn't has access to these hooks.
To save models we should write callback and encourage users to use it with our logger. But users can use native callbacks and control saving everything himself, and track it with DVC.